annotation-statistics/src/main/java/eu/europeana/annotation/statistics/service/AnnotationStatisticsService.java

```diff
@@ -0,0 +1,93 @@
+package eu.europeana.annotation.statistics.service;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import javax.annotation.Resource;
+
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.client.solrj.response.json.BucketJsonFacet;
+
+import eu.europeana.annotation.definitions.model.vocabulary.AnnotationScenarioTypes;
+import eu.europeana.annotation.solr.exceptions.AnnotationServiceException;
+import eu.europeana.annotation.solr.service.SolrAnnotationService;
+import eu.europeana.annotation.solr.vocabulary.SolrAnnotationConstants;
+import eu.europeana.annotation.statistics.model.AnnotationMetric;
+import eu.europeana.annotation.statistics.model.AnnotationStatistics;
+import eu.europeana.annotation.statistics.model.AnnotationStatisticsClient;
+import eu.europeana.annotation.statistics.model.AnnotationStatisticsUser;
+import eu.europeana.annotation.statistics.vocabulary.AnnotationStatisticsConstants;
+
+public class AnnotationStatisticsService {
+
+	@Resource
+	SolrAnnotationService solrService;
+
+    public void getAnnotationsStatistics(AnnotationMetric annoMetric) throws AnnotationServiceException {
+    	annoMetric.setTimestamp(new Date());  
+    	//getting the annotations statistics for the clients
+    	QueryResponse annoFacetStatsJson = solrService.getAnnotationStatisticsJsonNestedFacets(SolrAnnotationConstants.GENERATOR_URI);
```

Comment:
there is an open question in the ticket to clear out which is the main facet, the scenario or the generator (client application)

Reply:
I kept it the way it is defined in the specs.